src/system/ui/pages/term/index.js

```diff
@@ -1,40 +1,168 @@
 import React from 'react'
 import { __ } from '@wordpress/i18n'
-import { Page, Form } from 'ui'
+import { Page, Form, Button } from 'ui'
+import { getWpRest } from 'utils/wordpress'
+import { getSystemActions } from 'data'
 
-export const Term = ( { location } ) => {
+
+export const Term = ( { location, history } ) => {
 	const { item } = location.state
-	const { form, useFormContext } = Form.useFormData_Deprecated( {
-		title: {
-			label: __( 'Name' ),
-			labelPlacement: 'beside',
-		},
-		slug: {
-			label: __( 'Slug' ),
-			labelPlacement: 'beside',
-		},
-		parent: {
-			label: __( 'Parent' ),
-			labelPlacement: 'beside',
-			options: {
-				'': __( 'None' ),
+	const {
+		id,
+		title,
+		description,
+		slug,
+		parent,
+		count,
+		taxonomy
+	} = item
+	const wpRest = getWpRest()
+	const { setCurrentHistoryState } = getSystemActions()
+
+
+	const onSubmit = ( { changed, ids } ) => {
+
+		const data = {
+			meta: {},
+		}
+		for ( let key in changed ) {
+			if ( ! ids[key] ) {
+				continue
+			}
+			data[ids[key]] = changed[key]
+		}
+
+		if ( 'description' in changed ) {
+			item.description = changed.description
+		}
+
+		if ( 'title' in changed ) {
+			item.title = changed.title
+			data.name = changed.title
+		}
+
+		if ( 'parent' in changed ) {
+			item.parent = changed.parent
+			data.parent = changed.parent
+		}
+
+		if ( 'slug' in changed ) {
+			item.slug = changed.slug
+		}
+
+		const handleError = error => {
+			setIsSubmitting( false )
+			alert( __( 'Error: Changes not saved! Please try again.' ) )
+			if ( error ) {
+				console.log( error ) // eslint-disable-line no-console
+			}
+		}
+
+		wpRest.terms().update( item.id, 'data', data ).then( response => {
+			const { data } = response
+			if ( data.error ) {
+				handleError()
+			} else {
+				setCurrentHistoryState( { item } )
+				setIsSubmitting( false )
+				alert( __( 'Changes Saved!' ) )
+			}
+		} ).catch( error => {
+			handleError( error )
+		} )
+	}
+
+	const deleteTerm = () => {
+		if ( confirm( __( 'Do you really want to delete this term?' ) ) ) {
+			wpRest.terms().update( id, 'trash' ).then( () => {
+				alert( 'Term permanently deleted!' )
+			} )
+			history.goBack()
+		}
+	}
+
+
+	const { hasChanges, resetForm, setIsSubmitting, submitForm, renderForm } = Form.useForm( {
+
+
+		sections: {
+			details: {
+				label: __( 'Details' ),
+				fields: {
+					title: {
+						label: __( 'Name' ),
+						labelPlacement: 'above',
+						value: title
```

Comment:
Form config doesn't accept a value (you may have seen this in old code). To set the initial value, use the `defaults` object below.